src/main/asciidoc/reference/vault-secret-engines.adoc

```diff
@@ -0,0 +1,252 @@
+
+The token store can also be used to bypass any other auth method: you can create tokens directly, as well as perform a variety of other operations on tokens such as renewal and revocation.
+
+Spring Vault uses this backend to renew and revoke session tokens supplied by the configured <<vault.core.authentication, authentication method>>.
+
+The following examples shows how to request, renew and revoke a Vault token from within your application:
+
+====
+[source,java,indent=0]
+----
+include::../{example-root}/Token.java[tags=tokenApi]
+----
+<1> Create an token applying role defaults.
+<2> Using the builder API, you can define fine-grained settings for the token to request.
+Requesting a token returns a `VaultToken` which is used as value object for Vault tokens.
+<3> Tokens can be renewed through the Token API. Typically, that is done by `SessionManager` to keep track of the Vault session token.
+<5> Tokens can be revoked if needed through the Token API. Typically, that is done by `SessionManager` to keep track of the Vault session token.
+====
+
+You can find more details about the https://www.vaultproject.io/api-docs/auth/token[Vault Token Auth Method API] in the Vault reference documentation.
+
+[[vault.core.backends.transit]]
+== Transit Backend
+
+The transit secrets engine handles cryptographic functions on data in-transit.
+Vault doesn't store the data sent to this secrets engine.
+It can also be seen as "cryptography as a service" or "encryption as a service".
+The transit secrets engine can also sign and verify data; generate hashes and HMACs of data; and act as a random bytes source.
+
+The primary use case for transit is to encrypt data from applications while still storing that encrypted data in some primary data store.
+This relieves the burden of proper encryption/decryption from application developers and pushes the burden onto the operators of Vault.
+
+Spring Vault supports a wide range of Transit operations:
+
+* Key creation
+* Key reconfiguration
+* Encryption/Decryption/Rewrapping
+* HMAC computation
+* Signing and signature verification
+
+All operations within `transit` are centered around keys.
+The Transit engine supports the versioning of keys allowing you to pick a key type https://www.vaultproject.io/docs/secrets/transit[from a variety of key types].
+Note that depending on the key type may impose a limitation on which operations can used.
+
+Let's take a look at encryption and decryption.
+The following examples shows how to create a key and how to encrypt and decrypt data:
+
+====
+[source,java,indent=0]
+----
+include::../{example-root}/Transit.java[tags=encryptSimple]
+----
+<1> First, we need a key to begin with.
+Each key requires the type to be specified. `aes128-gcm96` supports encryption, decryption, key derivation, and convergent encryption of which we need encryption and decryption for this example.
```

Comment:
Add a comma after "convergent encryption" (so "convergent encryption, of which").